### PR TITLE
Tools: Replay: option to print packet counts out at end

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: Tools/Replay/Replay.cpp [hunk 1]
```diff
@@ -814,6 +843,10 @@ void Replay::loop()
         return;
     }
 
+    if (check_solution) {
+        report_checks();
+    }
```
Comment (OXINARF) [inline Tools/Replay/Replay.cpp]: Can you explain this change? It looks like it's a big difference to what was done before.

File: Tools/Replay/DataFlashFileReader.cpp [hunk 2]
```diff
@@ -15,7 +15,9 @@ uint64_t now() {
 }
 
 DataFlashFileReader::DataFlashFileReader() :
-    start_micros(now())
+    fd(-1),
```
Comment (OXINARF) [inline Tools/Replay/DataFlashFileReader.cpp]: Why? This is already initialized in the declaration.

File: Tools/Replay/Replay.cpp [hunk 3]
```diff
@@ -780,9 +787,31 @@ void Replay::flush_and_exit()
...
 
+void Replay::show_packet_counts()
+{
+    uint64_t counts[LOGREADER_MAX_FORMATS];
+    logreader.get_packet_counts(counts);
+    char format_type[5];
+    uint64_t total = 0;
+    for(uint16_t i=0;i<LOGREADER_MAX_FORMATS;i++) {
+        if (counts[i] != 0) {
+            // don't call format_type
```
[the remark this answers is not in the document]
Comment (peterbarker) [reply Tools/Replay/Replay.cpp]: Leftover; used to be an else clause involved.  Removed comment.

File: Tools/Replay/Replay.cpp (hunk 1 above)
Comment (peterbarker) [reply Tools/Replay/Replay.cpp]: Bad rebase.  Fixed.

File: Tools/Replay/DataFlashFileReader.cpp (hunk 2 above)
Comment (peterbarker) [reply Tools/Replay/DataFlashFileReader.cpp]: Can't be too careful.

Removed it and moved initialisation of packets counts into declaration.

**Comment by peterbarker** [comment]
@OXINARF Fresh version push up.  Rebase bit me hard on this one.  OTOH, Replay must be pretty stable; this patchset is out of nappies by now - I just hadn't ever PR'd it.

**Comment by OXINARF** [comment]
Merged, thanks!

**Pull request merged**